src/main/java/com/github/splendor_mobile_game/game/model/User.java

```diff
@@ -75,9 +75,18 @@ public void changeTokens(Map<TokenType, Integer> tokensChange) {
         }
     }
 
-    //method for buying cards
-    public void buyCard(Card card) throws NotEnoughTokensException {
 
+    // Returns how many goldenTokens must be used in order to buy a card
+
+    /**
+     * Returns number of golden tokens needed to buy a Card
+     *
+     * @param card Card which user wants to buy
+     * @return -1 user is not able to buy a card
+     *          0 no golden tokens are required to buy a card
+     *         >0 number of needed golden tokens
```

Comment:
Can we use enums with variants wrapping a value? Like `Result::USER_NOT_ABLE_TO_BUY_CARD`, `Result::GOLD_TOKENS_NOT_NEEDED` and `Result::NEEDED_GOLD_TOKENS(int n)`. Sorry for using `Rust` syntax and I don't particularly want you to make this implementation. Just asking if it's possible in Java xD

Reply:
Yeah, we can use enums with optional parameters, or we can just use String messgaes